internal/host/validator.go

```diff
@@ -1022,49 +855,44 @@ func allImagesValid(imageStatuses common.ImageStatuses) bool {
    Since all pre-install validations have to pass before starting installation, it is mandatory that in case installation
    on the current boot device has not been attempted yet, this validation must pass.
 */
-func (v *validator) sufficientOrUnknownInstallationDiskSpeed(c *validationContext) ValidationStatus {
+func (v *validator) sufficientOrUnknownInstallationDiskSpeed(c *validationContext) (ValidationStatus, string) {
 	info, err := v.getBootDeviceInfo(c.host)
 	if err != nil {
-		return ValidationError
+		return ValidationError, "Validation error"
 	}
-
-	return boolValue(info == nil || info.DiskSpeed == nil || !info.DiskSpeed.Tested || info.DiskSpeed.ExitCode == 0)
-}
-
-func (v *validator) printSufficientOrUnknownInstallationDiskSpeed(c *validationContext, status ValidationStatus) string {
-	switch status {
-	case ValidationSuccess:
+	if info == nil || info.DiskSpeed == nil || !info.DiskSpeed.Tested || info.DiskSpeed.ExitCode == 0 {
 		info, _ := v.getBootDeviceInfo(c.host)
```

Comment:
This is already being called as the first line of this function. You should be able to simplify this.